File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/HomeDetailsModulesQuickservers.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Dashboard module configuration for QuickServer services.
 */
@Schema(description = "Dashboard module configuration for QuickServer services.")
@Validated
@NotUndefined



public class HomeDetailsModulesQuickservers   {
  @JsonProperty("icon")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String icon = null;

  @JsonProperty("view_link")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String viewLink = null;

  @JsonProperty("heading")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String heading = null;

  @JsonProperty("buy_link")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String buyLink = null;

  @JsonProperty("list_link")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String listLink = null;


  public HomeDetailsModulesQuickservers icon(String icon) { 

    this.icon = icon;
    return this;
  }

  /**
   * The icon for quick servers.
   * @return icon
   **/
  
  @Schema(example = "database", description = "The icon for quick servers.")
  
  public String getIcon() {  
    return icon;
  }



  public void setIcon(String icon) { 
    this.icon = icon;
  }

  public HomeDetailsModulesQuickservers viewLink(String viewLink) { 

    this.viewLink = viewLink;
    return this;
  }

  /**
   * Link to view quick servers.
   * @return viewLink
   **/
  
  @Schema(example = "view_qs", description = "Link to view quick servers.")
  
  public String getViewLink() {  
    return viewLink;
  }



  public void setViewLink(String viewLink) { 
    this.viewLink = viewLink;
  }

  public HomeDetailsModulesQuickservers heading(String heading) { 

    this.heading = heading;
    return this;
  }

  /**
   * Heading for quick servers.
   * @return heading
   **/
  
  @Schema(example = "Quick Servers", description = "Heading for quick servers.")
  
  public String getHeading() {  
    return heading;
  }



  public void setHeading(String heading) { 
    this.heading = heading;
  }

  public HomeDetailsModulesQuickservers buyLink(String buyLink) { 

    this.buyLink = buyLink;
    return this;
  }

  /**
   * Link to order quick server.
   * @return buyLink
   **/
  
  @Schema(example = "order_quickserver", description = "Link to order quick server.")
  
  public String getBuyLink() {  
    return buyLink;
  }



  public void setBuyLink(String buyLink) { 
    this.buyLink = buyLink;
  }

  public HomeDetailsModulesQuickservers listLink(String listLink) { 

    this.listLink = listLink;
    return this;
  }

  /**
   * Link to view quick servers list.
   * @return listLink
   **/
  
  @Schema(example = "view_quickservers_list", description = "Link to view quick servers list.")
  
  public String getListLink() {  
    return listLink;
  }



  public void setListLink(String listLink) { 
    this.listLink = listLink;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    HomeDetailsModulesQuickservers homeDetailsModulesQuickservers = (HomeDetailsModulesQuickservers) o;
    return Objects.equals(this.icon, homeDetailsModulesQuickservers.icon) &&
        Objects.equals(this.viewLink, homeDetailsModulesQuickservers.viewLink) &&
        Objects.equals(this.heading, homeDetailsModulesQuickservers.heading) &&
        Objects.equals(this.buyLink, homeDetailsModulesQuickservers.buyLink) &&
        Objects.equals(this.listLink, homeDetailsModulesQuickservers.listLink);
  }

  @Override
  public int hashCode() {
    return Objects.hash(icon, viewLink, heading, buyLink, listLink);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class HomeDetailsModulesQuickservers {\n");
    
    sb.append("    icon: ").append(toIndentedString(icon)).append("\n");
    sb.append("    viewLink: ").append(toIndentedString(viewLink)).append("\n");
    sb.append("    heading: ").append(toIndentedString(heading)).append("\n");
    sb.append("    buyLink: ").append(toIndentedString(buyLink)).append("\n");
    sb.append("    listLink: ").append(toIndentedString(listLink)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/HomeDetailsModulesQuickservers.cpp


#include "HomeDetailsModulesQuickservers.h"

using namespace Tiny;

HomeDetailsModulesQuickservers::HomeDetailsModulesQuickservers()
{
	icon = std::string();
	view_link = std::string();
	heading = std::string();
	buy_link = std::string();
	list_link = std::string();
}

HomeDetailsModulesQuickservers::HomeDetailsModulesQuickservers(std::string jsonString)
{
	this->fromJson(jsonString);
}

HomeDetailsModulesQuickservers::~HomeDetailsModulesQuickservers()
{

}

void
HomeDetailsModulesQuickservers::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *iconKey = "icon";

    if(object.has_key(iconKey))
    {
        bourne::json value = object[iconKey];



        jsonToValue(&icon, value, "std::string");


    }

    const char *view_linkKey = "view_link";

    if(object.has_key(view_linkKey))
    {
        bourne::json value = object[view_linkKey];



        jsonToValue(&view_link, value, "std::string");


    }

    const char *headingKey = "heading";

    if(object.has_key(headingKey))
    {
        bourne::json value = object[headingKey];



        jsonToValue(&heading, value, "std::string");


    }

    const char *buy_linkKey = "buy_link";

    if(object.has_key(buy_linkKey))
    {
        bourne::json value = object[buy_linkKey];



        jsonToValue(&buy_link, value, "std::string");


    }

    const char *list_linkKey = "list_link";

    if(object.has_key(list_linkKey))
    {
        bourne::json value = object[list_linkKey];



        jsonToValue(&list_link, value, "std::string");


    }


}

bourne::json
HomeDetailsModulesQuickservers::toJson()
{
    bourne::json object = bourne::json::object();





    object["icon"] = getIcon();






    object["view_link"] = getViewLink();






    object["heading"] = getHeading();






    object["buy_link"] = getBuyLink();






    object["list_link"] = getListLink();



    return object;

}

std::string
HomeDetailsModulesQuickservers::getIcon()
{
	return icon;
}

void
HomeDetailsModulesQuickservers::setIcon(std::string icon)
{
	this->icon = icon;
}

std::string
HomeDetailsModulesQuickservers::getViewLink()
{
	return view_link;
}

void
HomeDetailsModulesQuickservers::setViewLink(std::string view_link)
{
	this->view_link = view_link;
}

std::string
HomeDetailsModulesQuickservers::getHeading()
{
	return heading;
}

void
HomeDetailsModulesQuickservers::setHeading(std::string heading)
{
	this->heading = heading;
}

std::string
HomeDetailsModulesQuickservers::getBuyLink()
{
	return buy_link;
}

void
HomeDetailsModulesQuickservers::setBuyLink(std::string buy_link)
{
	this->buy_link = buy_link;
}

std::string
HomeDetailsModulesQuickservers::getListLink()
{
	return list_link;
}

void
HomeDetailsModulesQuickservers::setListLink(std::string list_link)
{
	this->list_link = list_link;
}



